rust/src/json_serialize.rs

```diff
@@ -0,0 +1,996 @@
+use std::collections::{BTreeMap, VecDeque};
+use std::fmt::{Display, Formatter};
+use std::iter::FromIterator;
+
+use itertools::Itertools;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::ser::Error;
+use unicode_segmentation::UnicodeSegmentation;
+
+use crate::JsError;
+use crate::ledger::common::value::BigInt;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    Number(BigInt),
+    String(String),
+    Array(Vec<Value>),
+    Object(BTreeMap<String, Value>),
+}
+
+impl Display for Value {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.serialize_str(self.clone().to_string().map_err(|err| std::fmt::Error::custom(format!("Can't serialize {:?}", err)))?.as_str())
```

Comment:
basically if you can provide a string it will be interpreted / serialized as a string. so I use my `to_string()` to convert to string and then use the standard serialize_str. Not sure if it's the best approach, but it should work at least